tune_sklearn/utils.py

```diff
@@ -30,6 +40,40 @@ def check_warm_start_ensemble(estimator):
             and hasattr(estimator, "n_estimators") and is_ensemble_subclass)
 
 
+def check_error_warm_start(estimator, estimator_config):
+    early_stop_type = get_early_stop_type(estimator)
+    if (early_stop_type == EarlyStopping.WARM_START_ITER
+            and "max_iter" in estimator_config):
+        raise ValueError("tune-sklearn uses `max_iter` to warm "
+                         "start, so this parameter can't be "
+                         "set when warm start early stopping. ")
+    if (early_stop_type == EarlyStopping.WARM_START_ENSEMBLE
+            and "n_estimators" in estimator_config):
+        raise ValueError("tune-sklearn uses `n_estimators` to warm "
+                         "start, so this parameter can't be "
+                         "set when warm start early stopping. ")
+
+
+def get_early_stop_type(estimator, early_stopping):
+    if not early_stopping:
+        return EarlyStopping.NO_EARLY_STOP
+    can_partial_fit = check_partial_fit(estimator)
+    can_warm_start_iter = check_warm_start_iter(estimator)
+    can_warm_start_ensemble = check_warm_start_ensemble(estimator)
```

Comment:
Need pipeline checks here. It would suffice to just call those `check_` methods on the final pipeline estimator.

Reply:
I think this should be checking on the final pipeline estimator right? In tune_basesearch this helper method is called on `self.base_estimator`

Reply:
Yes, that is correct. `estimator.steps[-1][1]` if `estimator` is a pipeline.